library/src/interactor_impl.cxx

```diff
@@ -71,8 +75,101 @@ class interactor_impl::internals
 #endif
   }
 
+  //----------------------------------------------------------------------------
+  // Method defined to normalize the Z axis so all models are treated temporarily
+  // as Z-up axis models.
+  void zUpTransforms(vtkMatrix3x3* to, vtkMatrix3x3* from)
+  {
+    vtkRenderer* renderer =
+      this->VTKInteractor->GetRenderWindow()->GetRenderers()->GetFirstRenderer();
+    const double* up = renderer->GetEnvironmentUp();
+    const double* right = renderer->GetEnvironmentRight();
+    double fwd[3];
+    vtkMath::Cross(right, up, fwd);
+    const double m[9] = {
+      right[0], right[1], right[2], //
+      fwd[0], fwd[1], fwd[2],       //
+      up[0], up[1], up[2],          //
+    };
+    to->DeepCopy(m);
+    from->DeepCopy(m);
+    from->Transpose();
+  }
+
+  //----------------------------------------------------------------------------
+  // Set the view orbit position on the viewport.
+  enum class ViewType
+  {
+    VT_FRONT,
+    VT_BACK,
+    VT_RIGHT,
+    VT_LEFT,
+    VT_TOP,
+    VT_ISOMETRIC
+  };
+  static void SetViewOrbit(ViewType view, internals* self)
+  {
+    vtkNew<vtkMatrix3x3> toZup, fromZup;
+    self->zUpTransforms(toZup, fromZup);
+    camera& cam = self->Window.getCamera();
+    vector3_t up = { 0, 0, 1 };
+    point3_t pos = cam.getPosition();
+    point3_t foc = cam.getFocalPoint();
+    point3_t Axis, newPos;
+
+    /* convert coords to +Z up */
+    toZup->MultiplyPoint(pos.data(), pos.data());
+    toZup->MultiplyPoint(foc.data(), foc.data());
+
+    const double dx = foc[0] - pos[0];
+    const double dy = foc[1] - pos[1];
+    const double dz = foc[2] - pos[2];
+    double radius = sqrt(dx * dx + dy * dy + dz * dz);
+    switch (view)
+    {
+      case ViewType::VT_FRONT:
+        Axis = { 0, 0, 1 };
+        fromZup->MultiplyPoint(up.data(), up.data());
+        break;
+      case ViewType::VT_BACK:
+        Axis = { 0, 0, -1 };
+        fromZup->MultiplyPoint(up.data(), up.data());
+        break;
+      case ViewType::VT_RIGHT:
+        Axis = { 1, 0, 0 };
+        fromZup->MultiplyPoint(up.data(), up.data());
+        break;
+      case ViewType::VT_LEFT:
+        Axis = { -1, 0, 0 };
+        fromZup->MultiplyPoint(up.data(), up.data());
+        break;
+      case ViewType::VT_TOP:
+        Axis = { 0, 1, 0 };
+        break;
+      case ViewType::VT_ISOMETRIC:
+        Axis = { -1, 1, 1 };
+        fromZup->MultiplyPoint(up.data(), up.data());
+        break;
+    }
+
+    fromZup->MultiplyPoint(Axis.data(), Axis.data());
```

Comment:
The point of the zUp idea was to be able to hardcode the internal math assuming z points up, so the axis values here look off, for example `ViewType::VT_FRONT: Axis = { 0, 0, 1 }` should not have a non-zero z, and `fromZup->MultiplyPoint(Axis.data(), Axis.data())` feels wrong too, because you want to transform your hardcoded values _back to_ the renderer's orientation, not _from_ it.
Looks like that code could work because somehow 2 wrongs do make a right in this case :)

I think it should go something like this instead (untested but hopefully you get the idea enough to fix it):
```suggestion
    switch (view)
    {
      case ViewType::VT_FRONT:
        Axis = { 0, +1, 0 };
        break;
      case ViewType::VT_BACK:
        Axis = { 0, -1, 0 };
        break;
      case ViewType::VT_RIGHT:
        Axis = { +1, 0, 0 };
        break;
      case ViewType::VT_LEFT:
        Axis = { -1, 0, 0 };
        break;
      case ViewType::VT_TOP:
        Axis = { 0, 0, -1 };
        up = { -1, 0, 0 };
        break;
      case ViewType::VT_ISOMETRIC:
        Axis = { -1, -1, -1 };
        break;
    }
    fromZup->MultiplyPoint(up.data(), up.data());
```

All of that needs to be double checked, I meant to build and run locally to do it more thoroughly but I haven't had time yet so here's my educated guess / gut feeling from just looking at the PR instead.

Reply:
After a few tweaks, it worked. I'll now update this on my new commit to this PR. Thanks for the suggestion!